kotest-assertions-arrow-fx-coroutines/build.gradle.kts

```diff
@@ -0,0 +1,136 @@
+import org.gradle.api.tasks.testing.logging.TestLogEvent
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+
+kotlin {
+  explicitApi()
+
+  targets {
+    metadata()
+
+    jvm {
+      compilations.all {
+        kotlinOptions.jvmTarget = "1.8"
+      }
+    }
+
+    js(IR) {
+      browser()
+      nodejs()
+    }
+
+    linuxX64()
+
+    mingwX64()
+
+    iosArm32()
+    iosArm64()
+    iosSimulatorArm64()
+    iosX64()
+    macosArm64()
+    macosX64()
+    tvosArm64()
+    tvosSimulatorArm64()
+    tvosX64()
+    watchosArm32()
+    watchosArm64()
+    watchosSimulatorArm64()
+    watchosX64()
+    watchosX86()
+  }
+
+  sourceSets {
+    val commonMain by getting {
+      dependencies {
+        implementation("org.jetbrains.kotlin:kotlin-stdlib:1.6.20")
+        implementation("io.kotest:kotest-assertions-core:5.2.1")
+        implementation("io.kotest:kotest-framework-engine:5.2.1")
+        implementation("io.kotest:kotest-property:5.2.1")
+        api(projects.kotestAssertionsArrow)
+        api("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.1")
+        compileOnly("io.arrow-kt:arrow-fx-coroutines:1.0.1")
+      }
+    }
+
+    val jvmMain by getting {
+      dependsOn(commonMain)
+    }
+
+    val commonTest by getting {
+      dependsOn(commonMain)
```

Comment:
Hey! Coming from https://youtrack.jetbrains.com/issue/KT-52003/MPP:-%22Expected-function-has-no-actual-declaration%22-for-Native-ta

Please do not add dependsOn edges from *Test to *Main source sets! This is considered harmful! Source sets will get associated automatically. Make sure that you structure your test source set hierarchy also.

Reply:
Thanks Sebastian,
now I am getting this error from kotlinx-coroutines, for now I will remove the runBlocking in this PR. 

```shell
> Task :kotest-assertions-arrow-fx-coroutines:watchosX64Test
Uncaught Kotlin exception: kotlin.IllegalStateException: There is no event loop. Use runBlocking { ... } to start one.
Invalid connection: com.apple.coresymbolicationd
    at 0   test.kexe                           0x1058835ca        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 90 (/Users/teamcity3/buildAgent/work/c3a91df21e46e2c8/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:24:37)
    at 1   test.kexe                           0x10587d566        kfun:kotlin.Exception#<init>(kotlin.String?){} + 86 (/Users/teamcity3/buildAgent/work/c3a91df21e46e2c8/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:23:44)
    at 2   test.kexe                           0x10587d756        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 86 (/Users/teamcity3/buildAgent/work/c3a91df21e46e2c8/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:34:44)
    at 3   test.kexe                           0x10587dba6        kfun:kotlin.IllegalStateException#<init>(kotlin.String?){} + 86 (/Users/teamcity3/buildAgent/work/c3a91df21e46e2c8/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:70:44)
    at 4   test.kexe                           0x105a4c187        kfun:kotlinx.coroutines.takeEventLoop#internal + 391 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Dispatchers.kt:42:5)
    at 5   test.kexe                           0x105a4bc43        kfun:kotlinx.coroutines.OldDefaultExecutor#dispatch(kotlin.coroutines.CoroutineContext;kotlinx.coroutines.Runnable){} + 179 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Dispatchers.kt:47:9)
    at 6   test.kexe                           0x105a210f3        kfun:kotlinx.coroutines.internal#resumeCancellableWith__at__kotlin.coroutines.Continuation<0:0>(kotlin.Result<0:0>;kotlin.Function1<kotlin.Throwable,kotlin.Unit>?){0§<kotlin.Any?>} + 1091 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/internal/DispatchedContinuation.kt:282:64)
    at 7   test.kexe                           0x105a30d51        kfun:kotlinx.coroutines.intrinsics#startCoroutineCancellable__at__kotlin.coroutines.SuspendFunction1<0:0,0:1>(0:0;kotlin.coroutines.Continuation<0:1>;kotlin.Function1<kotlin.Throwable,kotlin.Unit>?){0§<kotlin.Any?>;1§<kotlin.Any?>} + 593 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/intrinsics/Cancellable.kt:30:74)
Child process terminated with signal 6: Abort trap
    at 8   test.kexe                           0x105a30f8d        kfun:kotlinx.coroutines.intrinsics#startCoroutineCancellable$default__at__kotlin.coroutines.SuspendFunction1<0:0,0:1>(0:0;kotlin.coroutines.Continuation<0:1>;kotlin.Function1<kotlin.Throwable,kotlin.Unit>?;kotlin.Int){0§<kotlin.Any?>;1§<kotlin.Any?>} + 333 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/intrinsics/Cancellable.kt:25:10)
    at 9   test.kexe                           0x1059d1a45        kfun:kotlinx.coroutines.CoroutineStart#invoke(kotlin.coroutines.SuspendFunction1<0:0,0:1>;0:0;kotlin.coroutines.Continuation<0:1>){0§<kotlin.Any?>;1§<kotlin.Any?>} + 325 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/CoroutineStart.kt:110:30)
    at 10  test.kexe                           0x1059bfebc        kfun:kotlinx.coroutines.AbstractCoroutine#start(kotlinx.coroutines.CoroutineStart;0:0;kotlin.coroutines.SuspendFunction1<0:0,1:0>){0§<kotlin.Any?>} + 156 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/AbstractCoroutine.kt:126:9)
    at 11  test.kexe                           0x105a49104        kfun:kotlinx.coroutines#runBlocking(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,0:0>){0§<kotlin.Any?>}0:0 + 1140 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:55:15)
    at 12  test.kexe                           0x105bed6b1        kfun:io.kotest.assertions.arrow.fx.coroutines#runBlocking(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,0:0>){0§<kotlin.Any?>}0:0 + 177 (/Users/ulti_mate/IdeaProjects/kotest-extensions-arrow/kotest-assertions-arrow-fx-coroutines/src/nativeMain/kotlin/io/kotest/assertions/arrow/fx/coroutines/runBlocking.kt:11:3)
    at 13  test.kexe                           0x105bde751        kfun:io.kotest.assertions.arrow.fx.coroutines#resource__at__io.kotest.core.TestConfiguration(arrow.fx.coroutines.Resource<0:0>){0§<kotlin.Any?>}0:0 + 385 (/Users/ulti_mate/IdeaProjects/kotest-extensions-arrow/kotest-assertions-arrow-fx-coroutines/src/commonMain/kotlin/io/kotest/assertions/arrow/fx/coroutines/Resource.kt:78:3)
    at 14  test.kexe                           0x105c0672a        kfun:io.kotest.assertions.arrow.fx.coroutines.ResourceSpec.<init>$lambda-13#internal + 810 (/Users/ulti_mate/IdeaProjects/kotest-extensions-arrow/kotest-assertions-arrow-fx-coroutines/src/commonTest/kotlin/io/kotest/assertions/arrow/fx/coroutines/ResourceSpec.kt:34:11)
    at 15  test.kexe                           0x105c06884        kfun:io.kotest.assertions.arrow.fx.coroutines.ResourceSpec.$<init>$lambda-13$FUNCTION_REFERENCE$3.invoke#internal + 68 (/Users/ulti_mate/IdeaProjects/kotest-extensions-arrow/kotest-assertions-arrow-fx-coroutines/src/commonTest/kotlin/io/kotest/assertions/arrow/fx/coroutines/ResourceSpec.kt:12:33)
    at 16  test.kexe                           0x105c068ea        kfun:io.kotest.assertions.arrow.fx.coroutines.ResourceSpec.$<init>$lambda-13$FUNCTION_REFERENCE$3.$<bridge-UNNN>invoke(-1:0){}#internal + 90 (/Users/ulti_mate/IdeaProjects/kotest-extensions-arrow/kotest-assertions-arrow-fx-coroutines/src/commonTest/kotlin/io/kotest/assertions/arrow/fx/coroutines/ResourceSpec.kt:12:33)
    at 17  test.kexe                           0x105aa0466        kfun:io.kotest.core.spec.style.StringSpec#<init>(kotlin.Function1<io.kotest.core.spec.style.StringSpec,kotlin.Unit>){} + 246 (/Users/runner/work/kotest/kotest/kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/spec/style/stringSpec.kt:28:7)
    at 18  test.kexe                           0x105bfe12f        kfun:io.kotest.assertions.arrow.fx.coroutines.ResourceSpec#<init>(){} + 63 (/Users/ulti_mate/IdeaProjects/kotest-extensions-arrow/kotest-assertions-arrow-fx-coroutines/src/commonTest/kotlin/io/kotest/assertions/arrow/fx/coroutines/ResourceSpec.kt:12:22)
    at 19  test.kexe                           0x105c190af        kfun:io.kotest.assertions.arrow.fx.coroutines.ResourceSpec.$<init>$lambda-13$FUNCTION_REFERENCE$3.$<bridge-UNNN>invoke(-1:0){}#internal + 75807 
    at 20  test.kexe                           0x105c485a6        Kotlin_initRuntimeIfNeeded + 1446 
    at 21  test.kexe                           0x105c44416        Init_and_run_start + 22 
    at 22  dyld                                0x106556f0f        0x0 + 4401229583 
    at 23  ???                                 0x1126e951d        0x0 + 4604204317 
```